File: Data Structures & Algorithms/sqrtx/submission-3.swift
class Solution {
    func mySqrt(_ x: Int) -> Int {
        var l = 0 
        var r = x

        while l <= r {
            let m = l + (r - l)  / 2
            if m * m == x {
                return m 
            } else if (m * m) > x {
                r = m - 1
            } else {
                l = m + 1
            }
        }
        print(l,  r)
        return r
    }
}

File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        r = len(nums) - 1
        l = 0 
        while l <= r: 
            m = l + (r - l) // 2
            mid = nums[m]

            if mid == target: 
                return m 
            elif mid > target: 
                r = m - 1
            else: 
                l = m + 1
        return -1